sdk/src/client/core.rs

```diff
@@ -107,24 +108,45 @@ impl ClientInner {
         // request the node info every time, so we don't create invalid transactions/blocks.
         #[cfg(target_family = "wasm")]
         {
-            let current_time = crate::client::unix_timestamp_now().as_secs() as u32;
-            if let Some(last_sync) = *self.last_sync.lock().await {
-                if current_time < last_sync {
-                    return Ok(self.network_info.read().await.clone());
-                }
+            self.fetch_network_info().await
+        }
+
+        #[cfg(not(target_family = "wasm"))]
+        Ok(if let Some(info) = &*self.network_info.read().await {
+            info.clone()
+        } else {
```

Comment:
This should have the timestamp logic, otherwise we will never update the network info from the node

Reply:
That's what it did before, and relied on sync to update it

Reply:
Ah nvm I thought this is the syncing code